File: app/src/main/java/com/mobitill/barandrestaurant/data/order/OrderRepositoryComponent.java
package com.mobitill.barandrestaurant.data.order;

import com.mobitill.barandrestaurant.ApplicationModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by andronicus on 5/16/2017.
 */

@Singleton
@Component(modules = {OrderRepositoryModule.class, ApplicationModule.class})
public interface OrderRepositoryComponent {
    OrderRepository orderRepository();
}

File: app/src/main/java/com/mobitill/barandrestaurant/data/order/OrderRepositoryComponent.kt
package com.mobitill.barandrestaurant.data.order

import com.mobitill.barandrestaurant.ApplicationModule
import dagger.Component
import javax.inject.Singleton

/**
 * Created by andronicus on 5/16/2017.
 */

@Singleton
@Component(modules = arrayOf(OrderRepositoryModule::class, ApplicationModule::class))
interface OrderRepositoryComponent {
    fun orderRepository(): OrderRepository
}
